pyasn1/codec/ber/decoder.py

```diff
@@ -138,27 +173,26 @@ def valueDecoder(self, substrate, asn1Spec,
                      tagSet=None, length=None, state=None,
                      decodeFun=None, substrateFun=None,
                      **options):
-        head, tail = substrate[:length], substrate[length:]
 
         if substrateFun:
             return substrateFun(self._createComponent(
                 asn1Spec, tagSet, noValue, **options), substrate, length)
 
-        if not head:
+        if isEmpty(substrate) or not length:
```

Comment:
I have a feeling that we should try to avoid calling functions in the hot spots - they are not entirely for free.

Reply:
Yep, that is right, the cost of `isEmpty` and `asStream` are responsible for most of the overhead.

Reply:
Just a side note: for that reason in pyasn1 we have a bunch of singleton and immutable objects that could be compared with `is` which is cheap.

Reply:
Thanks for the suggestion.